src/Mollie/WC/Plugin.php

```diff
@@ -276,11 +276,36 @@ function ($settings) {
 		self::schedulePendingPaymentOrdersExpirationCheck();
         self::registerFrontendScripts();
         wp_enqueue_style('mollie-gateway-icons');
+        self::maybeFixSubscriptions();
 
 		// Mark plugin initiated
 		self::$initiated = true;
     }
 
+    /**
+     * See MOL-322
+     */
+    public static function maybeFixSubscriptions(){
+        $fixSubscriptionsProcess = get_option('mollie_wc_fix_subscriptions', false);
+        $hasSubscriptionPlugin = class_exists( 'WC_Subscriptions_Order' );
+        if (!$fixSubscriptionsProcess && $hasSubscriptionPlugin) {
+            $subscriptions = wcs_get_subscriptions([]);
```

Comment:
I believe it makes sense to check for the `wcs_get_subscriptions` function above instead of the `WC_Subscriptions_Order` class. The `wcs_get_subscriptions` may be present in one version but not in another one (I had the same problem last year:) ).

Also, this function call lacks arguments. For example, by default only the [first 10 subscriptions will be retrieved](https://github.com/wp-premium/woocommerce-subscriptions/blob/master/wcs-functions.php#L431). 

It probably makes sense to use the [meta_query](https://github.com/wp-premium/woocommerce-subscriptions/blob/master/wcs-functions.php#L476) parameter if you want to get subscriptions with or without some meta keys. For example, like this:

```php
wcs_get_subscriptions(
 [
  ...
  'meta_query' => [
   [
    'key' => '_mollie_customer_id',
    'value' => 'cst_'
   ]
  ]
 ]
);
```
Just in case, here are some docs: 
https://developer.wordpress.org/reference/classes/WP_Query/parse_query/
https://developer.wordpress.org/reference/classes/wp_meta_query/



Another thing is related to the performance. If you may assume the number of subscriptions you need to update or check is too high, it makes sense to consider background processing. It may be useful to avoid possible timeouts and slow site loading. At the very least I suggest doing this after the `is_admin()` check so the site wouldn't be slowed for customers.